docker-setup/action.yml

```diff
@@ -0,0 +1,120 @@
+
+inputs:
+  DOCKERHUB_USERNAME:
+    description: "Dockerhub username used to read public images"
+    required: true
+  DOCKERHUB_TOKEN:
+    description: "Dockerhub token used to read public images"
+    required: true
+
+  GITHUB_TOKEN:
+    description: "Token from secrets.GITHUB_TOKEN"
+    required: true
+
+  AWS:
+    description: "If we should use AWS"
+    default: true
+
+  AWS_ACCESS_KEY_ID:
+    description: "AWS access key used to write to ECR"
+    required: false
+  AWS_SECRET_ACCESS_KEY:
+    description: "AWS secret access key used to write to ECR"
+    required: false
+  AWS_REGION:
+    description: "AWS region"
+    required: false
+  AWS_ROLE_TO_ASSUME:
+    description: "AWS config to use"
+    required: false
+
+  ecr_registry:
+    default: ''
+    description: "ECR registry prefix"
+  ecr_type:
+    default: 'ecr'
+    description: "ecr or ecr-public"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Login to Docker Hub
+      if: ${{ !github.event.pull_request.head.repo.fork }}
+      uses: docker/login-action@v2
+      with:
+        username: ${{ inputs.DOCKERHUB_USERNAME }}
+        password: ${{ inputs.DOCKERHUB_TOKEN }}
+
+    - name: Configure AWS
+      if: ${{ !github.event.pull_request.head.repo.fork && inputs.AWS }}
+      shell: bash
+      run: |
+        mkdir -p ~/.aws
+        cat << EOF > ~/.aws/config
+        [default]
+        region = ${{ inputs.AWS_REGION }}
+        role_session_name = CI-GHA-voltti
+        role_arn = ${{ inputs.AWS_ROLE_TO_ASSUME }}
+        source_profile = default
+        EOF
+
+        cat << EOF > ~/.aws/credentials
+        [default]
+        aws_access_key_id = ${{ inputs.AWS_ACCESS_KEY_ID }}
+        aws_secret_access_key = ${{ inputs.AWS_SECRET_ACCESS_KEY }}
+        EOF
+
+    - name: Set up buildkitd
+      shell: bash
+      run: |
+        docker run -d \
+          --privileged \
+          --name remote-buildkit \
+          -p 1234:1234 \
+          -v "${HOME}/.aws/config:/root/.aws/config:ro" \
+          -v "${HOME}/.aws/credentials:/root/.aws/credentials:ro" \
+          moby/buildkit:v0.11.5 \
+          --addr unix:///run/buildkit/buildkitd.sock \
+          --addr tcp://0.0.0.0:1234
+
+    - name: Cache buildx
+      id: cache-buildx
+      uses: actions/cache@v3
+      with:
+        path: ~/.docker/cli-plugins/docker-buildx
+        key: ${{ runner.os }}-cache-buildx
+
+    - name: Build buildx
+      if: steps.cache-buildx.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        docker buildx bake "https://github.com/docker/buildx.git"
+        mkdir -p ~/.docker/cli-plugins
+        mv ./bin/build/buildx ~/.docker/cli-plugins/docker-buildx
+
+    - name: Buildx version
+      shell: bash
+      run: |
+        docker buildx version
+
+    - name: Buildx create
+      shell: bash
+      id: buildx-create
+      run: |
+        docker buildx create --driver remote docker-container://remote-buildkit --use
+
+    - name: Login to ECR
```

Comment:
https://github.com/aws-actions/amazon-ecr-login

Reply:
Normal login is used so public can be handled easily. As we do not do it here, I'll change this.